features/steps/test_starter/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/__main__.py

```diff
@@ -10,6 +12,10 @@
 def main(*args, **kwargs):
```

Comment:
If the interactive mode works properly, should we change `main` to return the run dictionary so it can be integrated with other things?

> Running kedro run or Kedro's entrypoint does not return anything - [features/steps/test_starter/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/__main__.py](https://github.com/kedro-org/kedro/pull/1423/files#diff-d12bcf43ce078ab58f3a662c0368908c6fc9b2c9e3798c298ed3f98bc3f94fc8)

https://github.com/kedro-org/kedro/issues/3237

Reply:
I think we should definitely consider it, but I think it can be considered a breaking change if `main` suddenly has a return value. Perhaps it's better to discuss that suggestion a bit more in a separate PR?

Reply:
I think @idanov and @antonymilne agreed that this is a right change. It didn't get merged because the PR was too big and block by the other changes.

In this case, I don't believe it is a breaking change, because `main` will always exit program previously so no one can relies on the return value of `main` (which is `None`). If anything, the change that making `main` no longer `exit` immediately is a bigger breaking change.

We can move this to a separate discussion.

Reply:
Got it @noklam! And you're right that actually this is a bigger change than the one you're suggesting. I'd still like to introduce it in a separate PR, because it's not immediately related to fixing the Databricks/interactive env stuff.

Reply:
Both options seem reasonable to me. If there's clarity on what to return already, we can do it in the current PRs; otherwise, move it to new ones. Up to the author's preference.

Reply:
We already have https://github.com/kedro-org/kedro/issues/2681 for returning a value.